### ISSUE TITLE
Significant performance degradation when using dynamic Themes

### ISSUE THREAD

**Issue by owinter86** [tamagui/tamagui#1809]
### Current Behavior

When utilising dynamic themes (switching from light mode to dark mode for example) the performance is reduced significantly, this seems to stem from any changes to themes will result in a new commit/render for each element that is exposed to the theme, rather than the changes updated in a single batched commit.

The example shows 1000 Stack components that change background color when the theme is updated from dark to light and the profiling screens shots shows the issue described above.

### Background changing with inline styles and RN View component
https://github.com/tamagui/tamagui/assets/2920419/1fe2edb4-f2e6-4782-9737-38801e56c83a

<img width="761" alt="Screenshot 2023-10-27 at 07 57 14" src="https://github.com/tamagui/tamagui/assets/2920419/3f0fb4ae-eaaf-4e4d-9ee7-1233aa96911f">

### Background changing with Tamagui stack and dynamic theme
https://github.com/tamagui/tamagui/assets/2920419/c2d20e2b-1efd-4fd7-8737-74596bca4db1

<img width="761" alt="Screenshot 2023-10-27 at 07 56 59" src="https://github.com/tamagui/tamagui/assets/2920419/16df8326-3e75-4244-94ef-ac079e5510b6">

### Expected Behavior

Themes should allow for batched commits, and have the same behaviour if updating a prop via a state change.

### Tamagui Version

```markdown
1.75.4
```

### Platform (Web, iOS, Android)

```markdown
iOS and Android
```

### Reproduction

```javascript
import { Stack, TamaguiProvider, Theme } from "@tamagui/core";
import { useState } from "react";
import { Button, SafeAreaView, View } from "react-native";

const newArray = Array.from(Array(1000).keys());

import { config } from "@tamagui/config/v2";
import { createTamagui } from "@tamagui/core";

const tamaguiConfig = createTamagui({
  ...config,
  themes: {
    dark: {
      canvas: "red",
    },
    light: {
      canvas: "blue",
    },
  } as const,
});

// this makes typescript properly type everything based on the config
type Conf = typeof tamaguiConfig;
declare module "@tamagui/core" {
  interface TamaguiCustomConfig extends Conf {}
}

export default function App() {
  const [theme, setTheme] = useState<"light" | "dark">("light");

  const [type, setType] = useState<"Stack" | "View">("Stack");
  return (
    <TamaguiProvider config={tamaguiConfig}>
      <Theme name={theme}>
        <SafeAreaView style={{ flex: 1 }}>
          <Button
            onPress={() => setTheme(theme === "light" ? "dark" : "light")}
            title="Toggle Theme"
          />
          <Button
            onPress={() => setType(type === "Stack" ? "View" : "Stack")}
            title={
              type === "Stack"
                ? "Using Stack from Tamagui"
                : "Using View with inline styles"
            }
          />
          <View
            style={{
              rowGap: 4,
              flexDirection: "row",
              columnGap: 4,
              flexWrap: "wrap",
            }}
          >
            {type === "Stack"
              ? newArray.map((item) => (
                  <Stack
                    key={item}
                    backgroundColor={"$canvas"}
                    // backgroundColor={theme === "dark" ? "red" : "blue"}
                    height={50}
                    width={50}
                  />
                ))
              : newArray.map((item) => (
                  <View
                    key={item}
                    style={{
                      backgroundColor: theme === "dark" ? "red" : "blue",
                      height: 50,
                      width: 50,
                    }}
                  />
                ))}
          </View>
        </SafeAreaView>
      </Theme>
    </TamaguiProvider>
  );
}
```
```


### System Info

```markdown
System:
    OS: macOS 14.0
    CPU: (8) arm64 Apple M1
    Memory: 102.00 MB / 16.00 GB
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 18.18.0 - /usr/local/bin/node
    Yarn: 1.22.19 - ~/.yarn/bin/yarn
    npm: 9.8.1 - /usr/local/bin/npm
    pnpm: 8.9.2 - /usr/local/bin/pnpm
    Watchman: 2023.09.25.00 - /opt/homebrew/bin/watchman
  Browsers:
    Chrome: 118.0.5993.117
    Safari: 17.0
  npmPackages:
    @babel/core: ^7.20.0 => 7.23.2
    @tamagui/babel-plugin: ^1.75.4 => 1.75.4
    @tamagui/config: ^1.75.4 => 1.75.4
    @tamagui/core: ^1.75.4 => 1.75.4
    @types/react: ~18.2.14 => 18.2.33
    babel-plugin-transform-inline-environment-variables: ^0.4.4 => 0.4.4
    expo: ~49.0.15 => 49.0.16
    expo-status-bar: ~1.6.0 => 1.6.0
    react: 18.2.0 => 18.2.0
    react-dom: 18.2.0 => 18.2.0
    react-native: 0.72.6 => 0.72.6
    react-native-reanimated: ~3.3.0 => 3.3.0
    tamagui: ^1.75.4 => 1.75.4
    typescript: ^5.1.3 => 5.2.2
```

**Comment by ismaelcompsci** [comment]
I have the same issue. Is this something that happens in recent versions? I am using v1.75.1

**Comment by natew** [comment]
Have a fix for this ready, will cut alongside a couple others here soon. Lesson learned: avoid queueMicrotask, I think it saved some perf on the web but have now ran into quite a few issues with it.

**Comment by natew** [comment]
fixed v1.75.6

**Comment by kpoelhekke** [comment]
@natew awesome to see this get fixed so quickly 🚀
